src/states/joining_peer.rs

```diff
@@ -145,8 +145,9 @@ impl JoiningPeer {
                 src: Authority::Node(_),
                 dst: Authority::Node(_),
             } => {
-                self.peer_map_mut().insert(pub_id, conn_info);
-                self.send_direct_message(&pub_id, DirectMessage::ConnectionResponse);
+                self.peer_map_mut().insert(pub_id, conn_info.clone());
+                let p2p_node = P2pNode::new(pub_id, conn_info);
+                self.send_direct_message(&p2p_node, DirectMessage::ConnectionResponse);
```

Comment:
Do you need to construct the `P2pNode`? `send_direct_message` should work with raw `ConnetionInfo` too.

Reply:
Good catch, I'll just pass the connection info